### PR TITLE
Feature/statuses manual sorting

### PR DESCRIPTION
**Pull request by pzajaczkowski**

### CODE REVIEW

File: src/core/Codend.Application/Projects/Commands/CreateProject/CreateProjectCommand.cs [hunk 1]
```diff
@@ -58,18 +59,22 @@ public async Task<Result<Guid>> Handle(CreateProjectCommand request, Cancellatio
...
         var project = resultProject.Value;
 
-        var resultStatuses = DefaultTaskStatus.List
-            .Select(status => ProjectTaskStatus.Create(project.Id, status.Name))
+        // Create default statuses with positions.
+        var defaultStatusesList = DefaultTaskStatus.SortedList;
+        var statusesPositions = Lexorank.GetSpacedOutValuesBetween(defaultStatusesList.Count);
+        var i = 0;
+        var resultStatuses = defaultStatusesList
+            .Select(status => ProjectTaskStatus.Create(project.Id, status.Name, statusesPositions[i++]))
```
Comment (maksimowiczm) [inline src/core/Codend.Application/Projects/Commands/CreateProject/CreateProjectCommand.cs:67]: Linq enumerates with index as second argument (status, index) => {}

File: src/core/Codend.Application/Projects/Commands/CreateProject/CreateProjectCommand.cs [hunk 2]
```diff
@@ -58,18 +59,22 @@ public async Task<Result<Guid>> Handle(CreateProjectCommand request, Cancellatio
...
         {
             return result.ToResult();
         }
 
-        // Set To-Do as first project default status.
-        var defaultStatus = resultStatuses.FirstOrDefault(status => 
-            status.Value.Name.Value == nameof(DefaultTaskStatus.ToDo))?.Value;
+        // Set To-Do as project default status.
+        var statuses = resultStatuses.Select(r => r.Value).ToList();
+        var defaultStatus = statuses.FirstOrDefault(status => status.Name.Value == DefaultTaskStatus.ToDo.Name);
```
Comment (maksimowiczm) [inline src/core/Codend.Application/Projects/Commands/CreateProject/CreateProjectCommand.cs:77]: Maybe move it to extension method 💀

File: src/core/persistence/Codend.Persistence/Repositories/ProjectTaskStatusRepository.cs [hunk 3]
```diff
@@ -57,4 +60,14 @@ public Task<bool> StatusExistsWithStatusIdAsync(ProjectTaskStatusId statusId, Pr
                     cancellationToken);
         return exists;
     }
+
+    public Task<Lexorank?> GetLowestStatusInProjectPositionAsync(ProjectId projectId,
+        CancellationToken cancellationToken)
```
Comment (maksimowiczm) [inline src/core/persistence/Codend.Persistence/Repositories/ProjectTaskStatusRepository.cs:65]: Cool indentation 💀

File: src/core/Codend.Presentation/Controllers/ProjectTaskStatusController.cs [hunk 4]
```diff
@@ -108,7 +109,8 @@ public ProjectTaskStatusController(IMediator mediator) : base(mediator)
         [FromBody] UpdateProjectTaskStatusRequest request) =>
         await Resolver<UpdateProjectTaskStatusCommand>
             .IfRequestNotNull(request)
-            .ResolverFor(new UpdateProjectTaskStatusCommand(statusId.GuidConversion<ProjectTaskStatusId>(), request.Name))
+            .ResolverFor(new UpdateProjectTaskStatusCommand(statusId.GuidConversion<ProjectTaskStatusId>(),
+                request.Name))
```
Comment (maksimowiczm) [inline src/core/Codend.Presentation/Controllers/ProjectTaskStatusController.cs:113]: cool indentation 💀

File: src/core/Codend.Presentation/Controllers/ProjectTaskStatusController.cs [hunk 5]
```diff
@@ -129,4 +131,43 @@ public ProjectTaskStatusController(IMediator mediator) : base(mediator)
             .For(new GetProjectTaskStatusesQuery(projectId.GuidConversion<ProjectId>()))
             .Execute(query => Mediator.Send(query))
             .ResolveResponse();
+
+    /// <summary>
+    /// Moves status position between two given lexorank positions.
+    /// </summary>
+    /// <param name="projectId">Id of the project status belongs to.</param>
```
Comment (maksimowiczm) [inline src/core/Codend.Presentation/Controllers/ProjectTaskStatusController.cs:138]: English 💀

File: src/core/Codend.Domain/Repositories/IProjectTaskStatusRepository.cs [hunk 6]
```diff
@@ -18,7 +19,16 @@ public interface IProjectTaskStatusRepository
...
-    public Task<bool> StatusExistsWithNameAsync(string name, ProjectId projectId, CancellationToken cancellationToken);
-    
-    public Task<bool> StatusExistsWithStatusIdAsync(ProjectTaskStatusId statusId, ProjectId projectId, CancellationToken cancellationToken);
+    Task<bool> StatusExistsWithNameAsync(string name, ProjectId projectId, CancellationToken cancellationToken);
+
+    Task<bool> StatusExistsWithStatusIdAsync(ProjectTaskStatusId statusId, ProjectId projectId,
+        CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Searches for and returns projectTaskStatus with lowest position, where lowest mean highest lexicographical ranking (last status).
```
Comment (maksimowiczm) [inline src/core/Codend.Domain/Repositories/IProjectTaskStatusRepository.cs:28]: Doesn't look like it return projectTaskStatus. Typo?

File: src/core/Codend.Application/ProjectTaskStatuses/Commands/MoveProjectTaskStatus/MoveProjectTaskStatusCommand.cs [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+using Codend.Domain.Entities;
+using Codend.Domain.Repositories;
+using Codend.Shared.Infrastructure.Lexorank;
+using FluentResults;
+using static Codend.Domain.Core.Errors.DomainErrors.General;
+
+namespace Codend.Application.ProjectTaskStatuses.Commands.MoveProjectTaskStatus;
+
+/// <summary>
+/// Command used for moving sprint task to other position.
```
Comment (maksimowiczm) [inline src/core/Codend.Application/ProjectTaskStatuses/Commands/MoveProjectTaskStatus/MoveProjectTaskStatusCommand.cs:14]: sprint task 💀

**Review by maksimowiczm — APPROVED** [review]

File: src/core/Codend.Application/Projects/Commands/CreateProject/CreateProjectCommand.cs (hunk 2 above)
Comment (pzajaczkowski) [reply src/core/Codend.Application/Projects/Commands/CreateProject/CreateProjectCommand.cs:77]: nah, it is not going to be used anywhere else

**Pull request merged**